components/raftstore/src/store/peer.rs

```diff
@@ -2792,7 +2792,20 @@ where
                 }
             }
         }
-        Ok((min_m.unwrap_or(0), min_c.unwrap_or(0)))
+        let (mut min_m, min_c) = (min_m.unwrap_or(0), min_c.unwrap_or(0));
+        if min_m < min_c {
+            warn!(
+                "min_matched < min_committed, raft progress is inaccurate";
+                "region_id" => self.region_id,
+                "peer_id" => self.peer.get_id(),
+                "min_matched" => min_m,
+                "min_committed" => min_c,
+            );
+            // Reset `min_matched` to `min_committed`, since a raft log is committed,
+            // the peer should also has replicated it
```

Comment:
Why? For example, a former leader can have log conflict and `matched` is 0. Its committed index doesn't necessary to be the minimized value of other peers.

Reply:
Enlarge `min_matched` can reduce the entry size for merge check, also `min_matched` and `min_committed` are unnecessary belong to the same peer

Reply:
You miss my point. The reason why `min_matched` can be set to `min_committed` is because `committed_index` is reported by follower and guaranteed logs at the index and before it are also persisted. It implies follower's actual `matched` should not be less than `committed_index`. And actual `min_matched` should not less than `min_committed`. It has nothing to do with whether the log is committed.

Reply:
Yes, it is the same reason I reset `min_matched`, the "a raft log" I mentioned in the comment is referring to the one at `min_committed`, I update the comment to make it more clear, PTAL again